src/care/DebugPlugin.cpp

```diff
@@ -0,0 +1,134 @@
+#include "care/DebugPlugin.h"
+#include "chai/ExecutionSpaces.hpp"
+#include "care/CHAICallback.h"
+#include "care/PluginData.h"
+
+// Std library headers
+#if defined(CARE_DEBUG) && !defined(_WIN32)
+#include <execinfo.h>
+#endif // defined(CARE_DEBUG) && !defined(_WIN32)
+
+#include <unordered_set>
+
+namespace care{
+   void DebugPlugin::registerPlugin() {
+      static RAJA::util::PluginRegistry::add<care::DebugPlugin> L ("Debug plugin", "CARE plugin for debugging");
+   }
+
+   void DebugPlugin::preLaunch(const RAJA::util::PluginContext& p) {
+#if !defined(CHAI_DISABLE_RM)
+      // Prepare to record CHAI data
+      if (CHAICallback::isActive()) {
+         PluginData::clearActivePointers();
+
+#if defined(CARE_GPUCC) && defined(CARE_DEBUG)
+         GPUWatchpoint::setOrCheckWatchpoint<int>();
+#endif // defined(CARE_GPUCC) && defined(CARE_DEBUG)
+      }
+#endif // !defined(CHAI_DISABLE_RM)
+   }
+
+
+   void DebugPlugin::postLaunch(const RAJA::util::PluginContext& p) {
+#if !defined(CHAI_DISABLE_RM)
+	   chai::ExecutionSpace space;
+
+	   switch (p.platform) {
+         case RAJA::Platform::host:
+         	space = chai::CPU; break;
+#if defined(CHAI_ENABLE_CUDA)
+    	   case RAJA::Platform::cuda:
+      	   space = chai::GPU; break;
+#endif
+#if defined(CHAI_ENABLE_HIP)
+         case RAJA::Platform::hip:
+            space = chai::GPU; break;
+#endif
+         default:
+            space = chai::NONE;
+      }
+
+      if (CHAICallback::isActive()) {			
+         writeLoopData(space, PluginData::getFileName(), PluginData::getLineNumber());
+	
+         // Clear out the captured arrays
+         PluginData::clearActivePointers();
```

Comment:
I just realized that we only need to clear out active pointers in preLaunch. It doesn't have to happen again in postLaunch.